website/docs/docs/building-a-dbt-project/building-models/configuring-incremental-models.md

```diff
@@ -64,14 +67,27 @@ from raw_app_data.events
 
 :::tip Optimizing your incremental model
 
-For more complex incremental models that make use of CTEs, you should consider the impact of the position of the `is_incremental()` macro on query performance. On some warehouses, filtering your records early can vastly improve the run time of your query!
+For more complex incremental models that make use of Common Table Expressions (CTEs), you should consider the impact of the position of the `is_incremental()` macro on query performance. In some warehouses, filtering your records early can vastly improve the run time of your query!
 
 :::
 
 ### Defining a uniqueness constraint (optional)
-`unique_key` is an optional parameter for incremental models that specifies a field which should be unique within your model. If the unique key of an existing row in your target table matches one of your incrementally transformed rows, the existing row will be updated. This ensures that you don't have multiple rows in your target table for a single row in your source data.
 
-You can define `unique_key` in a configuration block at the top of your model. The `unique_key` should be a single field name that is present in your model definition. While some databases support using expressions (eg. `concat(user_id, session_number)`), this syntax is not universally supported, so is not recommended. If you do not have a single field that is unique, consider first creating such a field in your model.
+A `unique_key` is an optional parameter for incremental models that specifies a field that should be unique within your model. You can define `unique_key` in a configuration block at the top of your model.
+
+For a given row in a `source` table: 
+* if all unique_key identified columns are present in the target table, an update procedure takes place
+* if any one of the unique_key columns in a source row does not find a matching target table row, the entire row is inserted into the target table
```

Comment:
> an update procedure takes place

This differs based on whether the database uses `merge` or `delete+insert`. If the latter, it's not an `update` of the existing row, it's a `delete` of the existing row + insertion of the new row (with the same key)

I think we need to balance the simplicity of the presentation here, while also finding ways to explain nuances like:
- `unique_key` is used by the most common incremental strategies (`delete+insert` + `merge`), but not all. For instance, it's not used by the `insert_overwrite` strategy. We talk about incremental strategies further down; I recently put together [an explainer in our internal Notion](https://www.notion.so/dbtlabs/Incremental-Strategies-573356fb7b2344fe86d3dd3089b8e2b1).
- If `unique_key` is not specified, the behavior is simple "append-only": all rows returned by the model's SQL are inserted into the preexisting table, with no regard for whether they represent dupes. (In `dbt-spark`, this strategy is named `append` explicitly; in other adapters, this is just the behavior of `delete+insert` or `merge` when no `unique_key` is specified.

Reply:
@jtcohen6 I think I've addressed these concerns in this commit https://github.com/dbt-labs/docs.getdbt.com/pull/1298/commits/1d3548ebe3c5843b1fb3bface03ba33366d8c54a but please let me know if it needs more tweaking!